src/devices/video/nt7534.cpp

```diff
@@ -0,0 +1,355 @@
+	save_item(NAME(m_ddram));
+	save_item(NAME(m_nibble));
+	save_item(NAME(m_rs_state));
+	save_item(NAME(m_rw_state));
+	save_item(NAME(m_backup_column));
+	save_item(NAME(m_read_modify_write));
+	save_item(NAME(m_adc));
+	save_item(NAME(m_reverse));
+	save_item(NAME(m_entire_display_on));
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+
+void nt7534_device::device_reset()
+{
+	memset(m_ddram, 0x00, sizeof(m_ddram));
+
+	m_page       = 0;
+	m_column     = 0;
+	m_dr         = 0;
+	m_ir         = 0;
+	m_display_on = false;
+	m_entire_display_on = false;
+	m_direction  = 1;
+	m_adc        = true;
+	m_read_modify_write = false;
+	m_reverse  = false;
+	m_data_len   = 8;
+	m_nibble     = false;
+	m_rs_state   = 0;
+	m_rw_state   = 0;
+
+	set_busy_flag(1520);
+}
+
+
+//-------------------------------------------------
+//  device_timer - handler timer events
+//-------------------------------------------------
+
+void nt7534_device::device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr)
+{
+	switch (id)
+	{
+		case TIMER_BUSY:
+			m_busy_flag = false;
+			break;
+	}
+}
+
+
+//**************************************************************************
+//  HELPERS
+//**************************************************************************
+
+void nt7534_device::set_busy_flag(uint16_t usec)
+{
+	m_busy_flag = true;
+	m_busy_timer->adjust( attotime::from_usec( usec ) );
+}
+
+void nt7534_device::update_nibble(int rs, int rw)
+{
+	if (m_rs_state != rs || m_rw_state != rw)
+	{
+		m_rs_state = rs;
+		m_rw_state = rw;
+		m_nibble = false;
+	}
+
+	m_nibble = !m_nibble;
+}
+
+//**************************************************************************
+//  device interface
+//**************************************************************************
+
+uint32_t nt7534_device::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
+{
+	bitmap.fill(0, cliprect);
+
+	uint16_t px;
+	uint8_t py;
+	for (uint16_t x = 0; x < 132; x++){
+		if (m_adc){
+			px = x;
+		} else {
+			px = 131 - x;
+		}
+		for (uint8_t y = 0; y < 65; y++){
+			py = (y + m_display_start_line - 32) % 65;
+			uint8_t page = py/8;
+			bitmap.pix16(y, x) = BIT(m_ddram[page*132 + px], py%8);
+		}
+	}
+
+	return 0;
+}
+
+READ8_MEMBER(nt7534_device::read)
+{
+	switch (offset & 0x01)
+	{
+		case 0: return control_read(space, 0);
+		case 1: return data_read(space, 0);
+	}
+
+	return 0;
+}
+
+WRITE8_MEMBER(nt7534_device::write)
+{
+	switch (offset & 0x01)
+	{
+		case 0: control_write(space, 0, data);  break;
+		case 1: data_write(space, 0, data);     break;
+	}
+}
+
+WRITE8_MEMBER(nt7534_device::control_write)
+{
+	if (m_data_len == 4)
+	{
+		update_nibble(0, 0);
+
+		if (m_nibble)
+		{
+			m_ir = data & 0xf0;
+			return;
+		}
+		else
+		{
+			m_ir |= ((data >> 4) & 0x0f);
+		}
+	}
+	else
+	{
+		m_ir = data;
+	}
+
+        if (m_ir == 0xE2){
```

Comment:
Oh dear, a mix of Allman and K&R within a single function ;)

Also, inconsistent case in hex constants - could you please consistently use upper- or lower-case hex within the file?  (Either 0xCAFE or 0xcafe style, not a mix within a single source file).